tests/test_utils.py

```diff
@@ -41,6 +42,15 @@ def test_enable_keepalive(self):
                 with patch('sys.platform', platform):
                     self.assertIsNone(enable_keepalive(Mock(), 10, 5))
 
+    def test_shell_quote(self):
+        self.assertEqual(shell_quote('value'), 'value')
+        self.assertEqual(shell_quote('value with spaces'), "'value with spaces'")
```

Comment:
It should be skipped on windows.

Reply:
Ah, thanks! Will do that now.

Reply:
What about testing values with a single/double quotes in the beginning/middle/end of the string (and different permutations of these combinations)?

It is important, because the regex checking that the provided string is already quoted is wrong.